modules/rabbitmq/src/main/java/org/testcontainers/containers/RabbitMQContainer.java

```diff
@@ -0,0 +1,334 @@
+package org.testcontainers.containers;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import org.jetbrains.annotations.NotNull;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.MountableFile;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static java.lang.String.join;
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toSet;
+
+/**
+ * Testcontainer for RabbitMQ.
+ *
+ * @author Martin Greber
+ */
+public class RabbitMQContainer extends GenericContainer<RabbitMQContainer> {
+
+    /**
+     * The image defaults to the official RabbitmQ image: <a href="https://hub.docker.com/_/rabbitmq/">RabbitMQ</a>.
+     */
+    private static final String DEFAULT_IMAGE_NAME = "rabbitmq";
+    private static final String DEFAULT_TAG = "3.7-management-alpine";
+
+    private static final int DEFAULT_AMQP_PORT = 5672;
+    private static final int DEFAULT_AMQPS_PORT = 5671;
+    private static final int DEFAULT_HTTPS_PORT = 15671;
+    private static final int DEFAULT_HTTP_PORT = 15672;
+
+    private String adminPassword = "guest";
+    private String adminUsername = "guest";
+    private final List<List<String>> values = new ArrayList<>();
+
+    /**
+     * Creates a Testcontainer using the official Neo4j docker image.
+     */
+    public RabbitMQContainer() {
+        this(DEFAULT_TAG);
+    }
+
+    /**
+     * Creates a Testcontainer using a specific docker image.
+     *
+     * @param tag The docker image to use.
+     */
+    public RabbitMQContainer(String tag) {
+        super(DEFAULT_IMAGE_NAME + ":" + tag);
+
+        this.waitStrategy = Wait.
+                forLogMessage(".*Server startup complete.*", 1)
+                .withStartupTimeout(Duration.ofSeconds(60));
+    }
+
+    @Override
+    public Set<Integer> getLivenessCheckPortNumbers() {
+
+        return Stream.of(DEFAULT_AMQP_PORT, DEFAULT_AMQPS_PORT, DEFAULT_HTTP_PORT, DEFAULT_HTTPS_PORT)
+                .map(this::getMappedPort)
+                .collect(toSet());
+    }
+
+    @Override
+    protected void configure() {
+        addExposedPorts(DEFAULT_AMQP_PORT, DEFAULT_AMQPS_PORT, DEFAULT_HTTP_PORT, DEFAULT_HTTPS_PORT);
```

Comment:
could you please move it to the constructor? `configure` is for dynamic things like the admin password, everything else should be configured in the constructor, so that it can be overridden externally without extending the class

Reply:
Done